geopandas/array.py

```diff
@@ -118,14 +120,18 @@ def _shapely_to_geom(geom):
     """
     Convert external Shapely object to internal representation (PyGEOS or Shapely).
     """
-    if not compat.USE_PYGEOS:
+    if compat.USE_SHAPELY_20:
+        return geom
+    elif not compat.USE_PYGEOS:
         return geom
     else:
         return vectorized._shapely_to_pygeos(geom)
 
 
 def _is_scalar_geometry(geom):
-    if compat.USE_PYGEOS:
+    if compat.USE_SHAPELY_20:
+        return isinstance(geom, BaseGeometry)
+    elif compat.USE_PYGEOS:
```

Comment:
Since the shapely 2.0 and pre-2.0 code is the same, this could be simplified slightly to 

```Python
if compat.USE_PYGEOS:
    return isinstance(geom, (pygeos.Geometry, BaseGeometry))
else:
    return isinstance(geom, BaseGeometry)
```